### PR TITLE
feat: add IdeButton

### PR DESCRIPTION
**Pull request by asfordmatt**
Contributing "IdeButton" component from CDAI components

#### Changelog

```
A       packages/cdai/.storybook/.babelrc
A       packages/cdai/.storybook/Container.js
A       packages/cdai/.storybook/addons.js
A       packages/cdai/.storybook/config.js
A       packages/cdai/.storybook/styles.scss
A       packages/cdai/.storybook/webpack.config.js
A       packages/cdai/ComponentStatus.json
M       packages/cdai/package.json
A       packages/cdai/src/component_helpers/StorybookHelper.js
A       packages/cdai/src/component_helpers/StorybookHelper.spec.js
A       packages/cdai/src/components/IdeButton/IdeButton.js
A       packages/cdai/src/components/IdeButton/IdeButton.spec.js
A       packages/cdai/src/components/IdeButton/IdeButton.stories.js
A       packages/cdai/src/components/IdeButton/_IdeButton.scss
A       packages/cdai/src/components/IdeButton/index.js
A       packages/cdai/src/globals/js/settings.js
A       packages/cdai/src/globals/js/settings.spec.js
A       packages/cdai/src/globals/scss/_variables.scss
A       packages/cdai/src/index.js
A       packages/cdai/src/index.scss
```

**New**

- Added "IdeButton" component from CDAI components

#### Testing / Reviewing

Tested with unit tests and storybook

### CODE REVIEW

File: packages/cdai/.storybook/.babelrc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (lee-chase) [inline packages/cdai/.storybook/.babelrc:1]: Do we need a per package storybook?

File: packages/cdai/.storybook/styles.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
+$css--font-face: true;
```
Comment (lee-chase) [inline packages/cdai/.storybook/styles.scss:1]: Currently, each component (Security and Experimental) has an SCSS file for both the component and story. We should probably discuss how to proceed.

File: packages/cdai/src/globals/scss/_variables.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// NOTE @simon See https://www.carbondesignsystem.com/guidelines/layout/#breakpoints
```
Comment (lee-chase) [inline packages/cdai/src/globals/scss/_variables.scss:1]: Why do we not read breakpoint information from `/carbon/carbon/packages/layout/scss/_breakpoint.scss`?

**Review by lee-chase — COMMENTED** [review]
Some thoughts for discussion

File: packages/cdai/.storybook/.babelrc (hunk 1 above)
Comment (vladbalanescu) [reply packages/cdai/.storybook/.babelrc:1]: maybe not, but at this stage, for ease of transferring the components over we thought it made sense. When we afterwards merge all of these components into 1 central folder we can resume at having just 1 global storybook

File: packages/cdai/.storybook/styles.scss (hunk 2 above)
Comment (vladbalanescu) [reply packages/cdai/.storybook/styles.scss:1]: Again, this was just a copy paste from our previous spot

File: packages/cdai/src/components/IdeButton/IdeButton.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'carbon-components-react';
+import { idePrefix } from '../../globals/js/settings';
```
[the remark this answers is not in the document]
Comment (vladbalanescu) [reply packages/cdai/src/components/IdeButton/IdeButton.js:4]: For now, for the same reasons as above we thought it made sense to be package level, i.e. `cdai`

File: packages/cdai/src/globals/scss/_variables.scss (hunk 3 above)
Comment (vladbalanescu) [reply packages/cdai/src/globals/scss/_variables.scss:1]: good question, I'll let Matt answer this one

Comment (asfordmatt) [reply packages/cdai/src/globals/scss/_variables.scss:1]: no idea, that's what was already there

**Review by vladbalanescu — APPROVED** [review]
LGTM 👍 , great stepping stone for our old ide-components

**Review by lee-chase — APPROVED** [review]
Looks good to me, although we should probably add copyright headers to each file where missing.

**Pull request merged**